### DISCUSSION TITLE
Grouped options support

### DISCUSSION THREAD

**Discussion by josefarias** [josefarias/hotwire_combobox#88] [Ideas]
Continuing from https://github.com/josefarias/hotwire_combobox/issues/87 which I'm closing in favor of this.

I think supporting this makes sense. Open to ideas for what the API should be.

**Comment by edwinv** [comment]
We are using the [`grouped_options_for_select` Rails helper](https://api.rubyonrails.org/classes/ActionView/Helpers/FormOptionsHelper.html#method-i-grouped_options_for_select) for populating our current select boxes. This structure works well.

I like the way the current implementation accepts different types of options through ActiveRecord instances, hashes and arrays. It would be great to allow grouping for these types without introducing a new `grouped_` helper. 

1. For `ActiveRecord::Relation` types you ideally want to call a method in the instance to fetch the group. This could be a `to_combobox_group` method.
2. With Single hash, Array of strings and Array of arrays you don't have enough space in the data structure to introduce a group.
3. This leaves us with the Array of hashes, this hash could contain a `group` value to determine the group.
4. We could introduce a new structure **Hash with X**, in which X can be any of the above structures. The `keys` of the hash are the groups to displays. The values are the options to render, which can be a single hash, array of strings and array of arrays.

If the `combobox_options` helper accepts these structures, the async rendering would also work out of the box.

For rendering the groups, the basic version could render a partial that is not overwritable by the user, only stylable through CSS. For our use case, we don't need custom rendering of groups through the `render_in` functionality.

**Comment by josefarias** [comment]
Coming in the next release!

https://github.com/josefarias/hotwire_combobox/pull/114
